cluster-autoscaler/processors/nodegroupset/compare_nodegroups.go

```diff
@@ -105,6 +106,19 @@ func CreateGenericNodeInfoComparator(extraIgnoredLabels []string) NodeInfoCompar
 	}
 }
 
+func CreateGenericNodeInfoLabelComparator(labels []string) NodeInfoComparator {
+	return func(n1, n2 *schedulerframework.NodeInfo) bool {
+		includedLabels := make(map[string]bool)
+		for _, l := range labels {
+			includedLabels[l] = true
+		}
+		if !compareLabels([]*schedulerframework.NodeInfo{n1, n2}, includedLabels, make(map[string]bool)) {
+			return false
+		}
+		return IsCloudProviderNodeInfoSimilar(n1, n2, make(map[string]bool))
+	}
+}
+
```

Comment:
As suggested, I created a new `comparator` to accept `labels` instead of `extraIgnoredLabels`, and compare nodes based on the specified labels. This `--balancing-labels` arg here bypasses the `--balancing-ignored-labels` flag. This also ignores the cloud provider-specific basic ignored flags. This calls for some discussion.

Rather than creating a new `comparator`, I'd personally prefer to edit the signature of the current comparator creators to accept an extra argument for `labels` in addition to `extraIgnoredLabels`, and compare the nodes for matching labels specified in `labels` if `labels` is not empty. If check fails, then we return, else we proceed to call `IsCloudProviderNodeInfoSimilar`, which takes care of ignored labels.

@MaciekPytel please share your thoughts and review here.

Reply:
I think the second implementation idea doesn't change current behavior at all. Currently every label that is not on ignored-labels list will cause the node groups to compare as different. Checking `--balancing-labels` first doesn't change the ultimate result, it just fails the comparison slightly earlier.

My understanding of #3615 is that specifying `--balancing-labels` makes comparison treat any label not on the list as ignored-label (current implementation treats any label that is not on ignored list as "balancing labels"). I think that:
* Specifying both `--balancing-labels` and `--balancing-ignored-labels` makes no sense. What would be the behavior for any label not on either list?
    * (if you agree with this logic) we should add validation in main.go that prevents setting both flags at the same time.
* Ignoring cloudprovider labels seems irrelevant when using `--balancing-labels` (since we would be ignoring all labels not on the list). 
* On the implementation side - calling IsCloudProviderNodeInfoSimilar as-is will not work as far as I can see, since it already has the label check embedded. I think the easiest way is to extract the resources check to a separate function and call that new function if `--balancing-labels are specified`, but there are plenty of other ways to refactor and I don't feel particularly strongly about which is the right one.

Reply:
👍 to

- Making `--balancing-labels` and `--balancing-ignored-labels` mutually exclusive
- Ignoring cloudprovider labels irrelevant when using `--balancing-labels`
- Yes, we'll need to refactor a underlying implementation of `IsCloudProviderNodeInfoSimilar` as you said. Since this is my first issue on CA, I was a bit hesitant to do much refactoring. Thanks for asserting my thoughts.